### PR TITLE
Add IdeaHubModuleNotification component.

### PR DESCRIPTION
**Pull request by asvinb**
## Summary

Addresses issue #3523 

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/components/notifications/IdeaHubModuleNotification.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+	const isActive = useSelect( ( select ) => select( CORE_MODULES ).isModuleActive( 'idea-hub' ) );
+	const isItemDismissed = useSelect( ( select ) => select( CORE_USER ).isItemDismissed( NOTIFICATION_ID ) );
+	const adminReauthURL = useSelect( ( select ) => select( MODULES_IDEA_HUB )?.getAdminReauthURL() );
+
+	const handleOnDismiss = useCallback( async () => {
+		await dismissItem( NOTIFICATION_ID );
+	}, [ dismissItem ] );
+
+	if ( isActive || isItemDismissed || isItemDismissed === undefined ) {
```
Comment (tofumatt) [inline assets/js/components/notifications/IdeaHubModuleNotification.js:49]: Technically, can't [`isActive` also be in an `undefined`/"loading" state](https://github.com/google/site-kit-wp/blob/65e6954e56d56167b15e218146e0e93e6ed852e5/assets/js/googlesitekit/modules/datastore/modules.js#L677-L679)? Similar to `isItemDismissed`, we might want to render `null` if it's still loading that info as well.

**Comment by tofumatt** [comment]
@googlebot I consent.

**Review by tofumatt — CHANGES_REQUESTED** [review]
I suppose because we don't have a setup screen for Idea Hub yet (but do need one), the redirect that this ends up using is a bit awkward; it redirects to this URL because Idea Hub gets setup, but there's no activation notice. But that's because we don't yet have a setup screen for it so that's fine...

Screenshot 2021-07-09 at 17 30 46

I did notice that dismissing the notification has an error in the JS console we should fix:

https://user-images.githubusercontent.com/90871/125109759-aa075900-e0db-11eb-84c4-132d81b9b543.mov

My commit efc79ee24814aecf53dd3c4994398e38a5f8fe4b fixes that.

-----

Regarding changes needed, it looks like the [storybook story isn't loading](https://google.github.io/site-kit-wp/storybook/pull/3677/?path=/story/modules-idea-hub-notifications-modulenotification--ready), because a `GET` call is failing/isn't being mocked:

Screenshot 2021-07-09 at 17 35 08

Just one other thing I think needs tweaking here, which is checking for an `undefined` return value from the `isActive()` selector.

File: assets/js/components/notifications/IdeaHubModuleNotification.js (hunk 1 above)
Comment (asvinb) [reply assets/js/components/notifications/IdeaHubModuleNotification.js]: 👍

**Comment by asvinb** [comment]
Thanks for catching that @tofumatt Guess I didn't check the story again after merging the latest code from develop. 🤦 
PR has been updated. Can you check again?

**Review by tofumatt — APPROVED** [review]
Awesome, thanks! 👍🏻

**Pull request merged**